src/jobserver-posix.cc

```diff
@@ -89,11 +82,6 @@ class PosixJobserverClient : public Jobserver::Client {
           std::string("Error opening fifo for reading: ") + strerror(errno);
       return false;
     }
-    if (!IsFifoDescriptor(read_fd_)) {
-      *error = "Not a fifo path: " + fifo_path;
-      // Let destructor close read_fd_.
-      return false;
-    }
```

Comment:
While it make sense to support non-FIFO file paths in the context of `guildmaster`, please ensure that passing an invalid file path is still an error reported by this function, as this will help catch common misconfiguration errors.

Moreover, printing a simple warning when the path is not a FIFO might be useful.